Map TextureUtil

Sturmlilie opened the pull request:
Names were chosen to make the most sense to someone familiar with OpenGL.  
First time, please be gentle (◠︿◠✿)

Sturmlilie committed:
Map TextureUtil

liach commented on mappings/net/minecraft/client/texture/TextureUtil.mapping:
```diff
@@ -1 +1,51 @@
+		ARG 4 height
+	METHOD method_24962 (Ljava/io/InputStream;)Ljava/nio/ByteBuffer;
+		ARG 0 inputStream
+	METHOD method_24963 uploadImageBGRA (Ljava/nio/IntBuffer;II)V
```
```suggestion
	METHOD method_24963 uploadImageBgra (Ljava/nio/IntBuffer;II)V
```

liach commented on mappings/net/minecraft/client/texture/TextureUtil.mapping:
```diff
@@ -1 +1,51 @@
+		ARG 1 width
+		ARG 2 height
+	METHOD method_24959 allocate (IIII)V
+		COMMENT
```
```suggestion
```

liach commented:
Also mind edit this pr so it merges into `1.16-pre2` instead?

Sturmlilie commented:
@liach Thanks for the review, and sorry about the `Bgra`, I read the contribution guide afterwards but somehow didn't connect the dots :sweat_smile:

Sturmlilie commented:
Actually don't merge yet please, I messed up the `Bgra` method.

Sturmlilie commented:
The method was badly named, I interpreted the GL parameters wrong; `GL_BGRA` + `GL_UNSIGNED_INT_8_8_8_8_REV` = logical bit order is ARGB [1]. I completely removed the suffix because BGRA seemed like a curiosity; ARGB on the other hand is the standard hex format used everywhere else in the code too. Now it's fixed.

[1] https://stackoverflow.com/questions/7786187/opengl-texture-upload-unsigned-byte-vs-unsigned-int-8-8-8-8/7786405#7786405

Sturmlilie committed:
Fixup: Remove unused mappings not picked up by Enigma

Sturmlilie committed:
Fixup: Properly overload 'allocate'

As it turns out, now that the bogus mappings are gone,
Enigma will actually let me properly overload
This was a learning experience :tiny-potato:
